### PR TITLE
lua: Move to lua 5.4 and include by default v5

### PR DESCRIPTION
**Pull request by J0eJ0h**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/4776

Describe changes:

- Move to using lua 5.4
- Remove luajit support and --enable-luajit configure option
- Change --enable-lua option to --disable-lua and build with lua by default
- CI Fixes

https://github.com/OISF/suricata/pull/10255 with:

- All internal references to luajit removed.

### CODE REVIEW

File: .github/workflows/builds.yml [hunk 1]
```diff
@@ -1559,7 +1559,7 @@ jobs:
           chmod 755 $HOME/.cargo/bin/cbindgen
           echo "$HOME/.cargo/bin" >> $GITHUB_PATH
       - run: ./autogen.sh
-      - run: ./configure --with-gnu-ld --enable-fuzztargets --disable-shared --enable-gccprotect
+      - run: ./configure --with-gnu-ld --enable-fuzztargets --disable-shared  --enable-gccprotect
```
Comment (jlucovsky) [inline .github/workflows/builds.yml:1562]: nit: extra whitespace after `--disable-shared`

**Review by jlucovsky — COMMENTED** [review]
This standardizes on lua5.4 -- I think that should be mentioned in the release notes or somewhere more appropriate.

**Review by catenacyber — CHANGES_REQUESTED** [review]
Setting the Github status as it needs a rebase, and there are comments to be taken care of

**Comment by jasonish** [comment]
Continued at https://github.com/OISF/suricata/pull/10801.

**Pull request closed without merging**